### PR TITLE
Cleanup oauth2 admin settings

### PR DESCRIPTION
**Pull request by CarlSchwan**
- Use more vue components
- Add link to doc

### CODE REVIEW

File: apps/oauth2/tests/Settings/AdminTest.php [hunk 1]
```diff
@@ -26,7 +26,8 @@
 use OCA\OAuth2\Db\ClientMapper;
 use OCA\OAuth2\Settings\Admin;
 use OCP\AppFramework\Http\TemplateResponse;
-use OCP\IInitialStateService;
+use OCP\IInitialState;
```
Comment (Pytal) [inline apps/oauth2/tests/Settings/AdminTest.php:29]: Could we drop the deprecated `OCP\IInitialStateService` and use the new `OCP\AppFramework\Services\IInitialState` instead? And if we do then could we also change the `IInitialStateService`'s below to `IInitialState`?

File: apps/oauth2/src/App.vue [hunk 2]
```diff
@@ -20,11 +20,9 @@
   -
   -->
 <template>
-	<div id="oauth2" class="section">
-		<h2>{{ t('oauth2', 'OAuth 2.0 clients') }}</h2>
-		<p class="settings-hint">
-			{{ t('oauth2', 'OAuth 2.0 allows external services to request access to {instanceName}.', { instanceName: OC.theme.name}) }}
-		</p>
+	<SettingsSection :title="t('oauth2', 'OAuth 2.0 clients')"
+		:description="t('oauth2', 'OAuth 2.0 allows external services to request access to {instanceName}.', { instanceName: OC.theme.name})"
```
Comment (Pytal) [inline apps/oauth2/src/App.vue:24]: Could we use `getCapabilities().theming.name` instead of `OC.theme.name`?

File: apps/oauth2/src/App.vue [hunk 3]
```diff
@@ -20,11 +20,9 @@
   -
   -->
 <template>
-	<div id="oauth2" class="section">
-		<h2>{{ t('oauth2', 'OAuth 2.0 clients') }}</h2>
-		<p class="settings-hint">
-			{{ t('oauth2', 'OAuth 2.0 allows external services to request access to {instanceName}.', { instanceName: OC.theme.name}) }}
-		</p>
+	<SettingsSection :title="t('oauth2', 'OAuth 2.0 clients')"
+		:description="t('oauth2', 'OAuth 2.0 allows external services to request access to {instanceName}.', { instanceName: OC.getCapabilities().theming.name })"
```
Comment (Pytal) [inline apps/oauth2/src/App.vue:24]: Could we call `getCapabilities()` from the https://github.com/nextcloud/nextcloud-capabilities package?

Comment (CarlSchwan) [reply apps/oauth2/src/App.vue:24]: Oh nice, I didn't knew about it

Comment (CarlSchwan) [reply apps/oauth2/src/App.vue]: done

File: apps/oauth2/tests/Settings/AdminTest.php [hunk 4]
```diff
@@ -44,10 +45,10 @@ class AdminTest extends TestCase {
 	protected function setUp(): void {
 		parent::setUp();
 
-		$this->initialStateService = $this->createMock(IInitialStateService::class);
+		$this->initialState = $this->createMock(IInitialState::class);
```
Comment (Pytal) [inline apps/oauth2/tests/Settings/AdminTest.php:48]: Could we rename the `IInitialStateService` and `initialStateService` above this line accordingly?

File: apps/oauth2/src/components/OAuthItem.vue [hunk 5]
```diff
@@ -42,14 +42,29 @@
 			</table>
 		</td>
 		<td class="action-column">
-			<span><a class="icon-delete has-tooltip" :title="t('oauth2', 'Delete')" @click="$emit('delete', id)" /></span>
+			<Button type="tertiary-no-background"
+				aria-label="t('oauth2', 'Delete')"
```
Comment (Pytal) [inline apps/oauth2/src/components/OAuthItem.vue:46]:
```suggestion
				:aria-label="t('oauth2', 'Delete')"
```

**Review by artonge — APPROVED** [review]

**Review by Pytal — APPROVED** [review]

**Pull request merged**